src/SFML/Graphics/Shader.cpp

```diff
@@ -156,11 +210,50 @@ Shader::CurrentTextureType Shader::CurrentTexture;
 
 
 ////////////////////////////////////////////////////////////
+struct Shader::UniformBinder : private sf::NonCopyable
+{
+    ////////////////////////////////////////////////////////////
+    /// \brief Constructor: set up state before uniform is set
+    ///
+    ////////////////////////////////////////////////////////////
+    UniformBinder(Shader* shader, const std::string& name) :
+    program(0),
+    location(-1)
+    {
+        if (shader->m_shaderProgram)
+        {
+            ensureGlContext();
+
+            // Enable program
+            glCheck(program = GLEXT_glGetHandle(GLEXT_GL_PROGRAM_OBJECT));
+            glCheck(GLEXT_glUseProgramObject(castToGlHandle(shader->m_shaderProgram)));
```

Comment:
An optimization I implemented a long time ago in 3DEE was to check if the currently bound program is actually different to the shader's program. If it isn't, there is no reason to bind it. Additionally, if the shader was already bound before this object is created, there is no reason for the "unbind" in the destructor. You could store 0 in the `program` member for this.

Doing this, you save a lot of unnecessary binds/unbinds when setting a lot of uniforms in series when the shader is already bound by `RenderTarget`.